src/cljs/main/broadfcui/page/workspace/method_configs/method_config_editor.cljs

```diff
@@ -200,9 +200,12 @@
         (create-section-header "Method Configuration Name")
         (create-section
          (if editing?
-           (style/create-text-field {:ref "confname" :style {:maxWidth 500}
-                                     :data-test-id "edit-method-config-name-input"
-                                     :defaultValue (:name config)})
+           [:div {}
+            (style/create-text-field {:ref "confname" :style {:width "100%"}
+                                      :data-test-id "edit-method-config-name-input"
+                                      :defaultValue (:name config)
+                                      :onKeyDown (common/create-key-handler [:space] #(.preventDefault %))})
```

Comment:
check out the `TextField` component as a full replacement for `create-text-field` (it's actually a wrapper). `TextField` has support for predicates. So instead of making a custom key handler that ignores spaces, you can just pull in some predicates. Something like the following should work:
```
            [input/TextField {:data-test-id "edit-method-config-name-input"
                              :ref "confname" :style {:autoFocus true :width "100%"}
                              :defaultValue (:name config)
                              :predicates [(input/nonempty "Config name")
                                           (input/alphanumeric_-period "Config name")]}]
```
I'm having problems running right now